### PR TITLE
Fix delegation double vote

### PR DESCRIPTION
**Pull request by bbenligiray**
Prevents delegations or undelegations from allowing double voting by:
- Not allowing the user to delegate if they have voted/created a proposal in the last week
- Not allowing the user to vote/create a proposal if they have undelegated in the last week

As a side effect, the primary and secondary voting apps are now using the same cooldown timer (i.e., the user can't create two proposals in a single week, one on the primary voting app and one on the secondary voting app). I think this is preferable anyway, it was weird for that to be allowed.

### CODE REVIEW

File: packages/pool/contracts/GetterUtils.sol [hunk 1]
```diff
@@ -277,6 +277,32 @@ abstract contract GetterUtils is StateUtils, IGetterUtils {
...
+        User storage user = users[userAddress];
+        unstaked = user.unstaked;
+        vesting = user.vesting;
+        unstakeScheduledFor = user.unstakeScheduledFor;
+        unstakeAmount = user.unstakeAmount;
+        mostRecentProposalTimestamp = user.mostRecentProposalTimestamp;
+        mostRecentVoteTimestamp = user.mostRecentVoteTimestamp;
+        mostRecentDelegationTimestamp = user.mostRecentDelegationTimestamp;
+        mostRecentUndelegationTimestamp = user.mostRecentUndelegationTimestamp;
+    }
```
Comment (BritikovKI) [inline packages/pool/contracts/GetterUtils.sol:304]: Hmmm, maybe It'll be good to use explicit return here?

**Review by BritikovKI — CHANGES_REQUESTED** [review]

File: packages/api3-voting/contracts/Api3Voting.sol [hunk 2]
```diff
@@ -257,8 +256,10 @@ contract Api3Voting is IForwarder, AragonApp {
         internal
         returns (uint256 voteId)
     {
-        require(userAddressToLastNewProposalTimestamp[msg.sender].add(api3Pool.EPOCH_LENGTH()) < now, "API3_HIT_PROPOSAL_COOLDOWN");
-        userAddressToLastNewProposalTimestamp[msg.sender] = now;
+        (, , , , uint256 mostRecentProposalTimestamp, , , uint256 mostRecentUndelegationTimestamp) = api3Pool.getUser(msg.sender);
```
[the remark this answers is not in the document]
Comment (BritikovKI) [reply packages/api3-voting/contracts/Api3Voting.sol:259]: My bad, it should be undelegation

File: packages/pool/contracts/GetterUtils.sol (hunk 1 above)
Comment (bbenligiray) [reply packages/pool/contracts/GetterUtils.sol:304]: You mean like a single
```
return;
```
at the end?

Comment (BritikovKI) [reply packages/pool/contracts/GetterUtils.sol:304]: Yep

Comment (bbenligiray) [reply packages/pool/contracts/GetterUtils.sol:304]: That would be extremely redundant

**Pull request merged**